### ISSUE TITLE
Dataclass member defaults may be shared by more than one instance

### ISSUE THREAD

**Issue by mvnmgrx** [mvnmgrx/kiutils#32]
I've noticed that default arguments for mutable members of a dataclass may be shared through more than one instance when doing something like this:
```python
from dataclasses import dataclass

@dataclass
class SubClass():
    some_pretty_int: int = 0

@dataclass
class MainClass():
    some_value: SubClass() = SubClass()

obj1 = MainClass()
obj2 = MainClass()

# Check identity
print(f'{id(obj1.some_value)}')
print(f'{id(obj2.some_value)}')

# Change one instance's value reflects to the other instance as well
obj2.some_value.some_pretty_int = 7
print(obj1.some_value)
print(obj2.some_value)
```

In this code, `obj1` and `obj2` share the same reference to `some_value`, as seen by the MWE's output:
```bash
PS E:\.....> py mwe.py
2100960016608
2100960016608
SubClass(some_pretty_int=7)            # Expected to be 0
SubClass(some_pretty_int=7)
```

**Possible workaround:**
Create mutable members with default values using the `field` directive with a lamda function:

```python
@dataclass
class MainClass():
    some_value: SubClass() = field(default_factory=lambda: SubClass())
```

This will yield the desired functionality:
```bash
PS E:\.....> py mwe.py
2393500270256
2393500269776
SubClass(some_pretty_int=0)
SubClass(some_pretty_int=7)
```

Although this wasn't an issue in `kiutils` yet, it uses the same programming style as described above. 

It needs to be checked if this could be come a problem.

**Comment by mvnmgrx** [comment]
Tested `kiutils` with Python 3.11 today. Using mutable default values for dataclass fields will now yield an exception:
```
ValueError: mutable default <class 'kiutils.items.common.ColorRGBA'> for field color is not allowed: use default_factory
```

Every mutable default value has to be replaced with a call to default_factory as shown above in order for Python 3.11 to be supported. I am still uncertain why this did not show any unwanted behavior in the past ..

**Comment by mvnmgrx** [comment]
Fixed in #35
